integration/image/list_test.go

```diff
@@ -57,6 +57,53 @@ func TestImagesFilterMultiReference(t *testing.T) {
 	}
 }
 
+func TestImagesFilterUntil(t *testing.T) {
+	ctx := setupTest(t)
+
+	client := testEnv.APIClient()
+
+	name := strings.ToLower(t.Name())
+
+	imgs := make([]string, 5)
+	var olderUntil string
+	var laterUntil string
+	for i := 0; i < 5; i++ {
+		if i > 0 {
+			time.Sleep(time.Millisecond)
+		}
+		ctrName := fmt.Sprintf("%s%d", name, i)
+		ctr := container.Create(ctx, t, client, container.WithName(ctrName))
+		id, err := client.ContainerCommit(ctx, ctr, containertypes.CommitOptions{Reference: ctrName})
+		assert.NilError(t, err)
+		imgs[i] = id.ID
+		if i == 3 {
+			res, err := client.ContainerInspect(ctx, ctr)
+			assert.NilError(t, err)
+			olderUntil = res.Created
+		} else if i == 4 {
+			res, err := client.ContainerInspect(ctx, ctr)
+			assert.NilError(t, err)
+			laterUntil = res.Created
+		}
```

Comment:
The `Created` timestamp of an image is the timestamp of when the image was created. Images created by committing a container are no exception. The `Created` timestamp of the source container is irrelevant information. (Think about it. The container could have been created days, months, or years in the past. And containers are mutable: images of the same container committed at different times could be vastly different in content.) **The `olderUntil` and `laterUntil` timestamps need to be (derived from) the `Created` timestamps of the images themselves to ensure the test is not flaky.** As there is no correlation between the image timestamps and any other timestamp aside from the order of operations of the test itself, using any other timestamps is not much better than just calling `time.Now()` directly at the right points in the test.

For the same reason, only one source container is needed, just like in `TestImagesFilterBeforeSince`.

You only need the image ID to query its `Created` timestamp. Since the committed images' IDs are assigned to the `imgs` slice, the code to fetch the `olderUntil` and `laterUntil` values [can be moved to below the loop.](https://en.wikipedia.org/wiki/Loop-switch_sequence)